reflect.go

```diff
@@ -54,13 +54,13 @@ type Schema struct {
 	PropertyNames        *Schema                                 `json:"propertyNames,omitempty"`        // section 10.3.2.4
 	// RFC draft-bhutton-json-schema-validation-00, section 6
 	Type              string              `json:"type,omitempty"`              // section 6.1.1
-	Enum              []interface{}       `json:"enum,omitempty"`              // section 6.1.2
-	Const             interface{}         `json:"const,omitempty"`             // section 6.1.3
-	MultipleOf        int                 `json:"multipleOf,omitempty"`        // section 6.2.1
-	Maximum           int                 `json:"maximum,omitempty"`           // section 6.2.2
-	ExclusiveMaximum  bool                `json:"exclusiveMaximum,omitempty"`  // section 6.2.3
-	Minimum           int                 `json:"minimum,omitempty"`           // section 6.2.4
-	ExclusiveMinimum  bool                `json:"exclusiveMinimum,omitempty"`  // section 6.2.5
+	Enum              []any               `json:"enum,omitempty"`              // section 6.1.2
+	Const             any                 `json:"const,omitempty"`             // section 6.1.3
+	MultipleOf        *json.Number        `json:"multipleOf,omitempty"`        // section 6.2.1
```

Comment:
Do these need to be pointers? `json.Number` is a type of `string`, so the omitempty flag should remove them. They'll be slightly easier to use if so. Also, I think this same pattern could be applied to the `MaxLength` and `MinLength` properties (perhaps more) for the sake of consistency.

Reply:
Moved away from pointers in 372fbaa

Reply:
For the `minLength` & `maxLength` we have different restrictions, see https://json-schema.org/draft/2020-12/json-schema-validation.html#section-6.3.1 & https://json-schema.org/draft/2020-12/json-schema-validation.html#section-6.3.2 (they are non-negative integers)